### ISSUE TITLE
[Feature] Add permission setting on who can upload files

### ISSUE THREAD

**Issue by maxpen** [RocketChat/Rocket.Chat#8427]
### Server Setup Information:

* Version of Rocket.Chat Server:
* Operating System:
* Deployment Method(snap/docker/tar/etc):
* Number of Running Instances: 
* DB Replicaset Oplog:
* Node Version:

### Steps to Reproduce:

1. 
2. 
3.

**Comment by fais3000** [comment]
@MartinSchoeler @maxpen  Agreed, specially helpful when Anonymous write is on and we have Anonymous role.
